tests/integration/test_dynamodb.py

```diff
@@ -799,27 +799,34 @@ def test_dynamodb_stream_records_with_update_item(self, dynamodb):
                 ReturnConsumedCapacity="INDEXES",
             )
 
-        records = ddbstreams.get_records(ShardIterator=iterator_id)
-        assert records["ResponseMetadata"]["HTTPStatusCode"] == 200
-        assert len(records["Records"]) == 2
-        assert isinstance(
-            records["Records"][0]["dynamodb"]["ApproximateCreationDateTime"],
-            datetime,
-        )
-        assert records["Records"][0]["dynamodb"]["ApproximateCreationDateTime"].microsecond == 0
-        assert records["Records"][0]["eventVersion"] == "1.1"
-        assert records["Records"][0]["eventName"] == "INSERT"
-        assert "OldImage" not in records["Records"][0]["dynamodb"]
-        assert int(records["Records"][0]["dynamodb"]["SequenceNumber"]) > starting_sequence_number
-        assert isinstance(
-            records["Records"][1]["dynamodb"]["ApproximateCreationDateTime"],
-            datetime,
-        )
-        assert records["Records"][1]["dynamodb"]["ApproximateCreationDateTime"].microsecond == 0
-        assert records["Records"][1]["eventVersion"] == "1.1"
-        assert records["Records"][1]["eventName"] == "MODIFY"
-        assert "OldImage" in records["Records"][1]["dynamodb"]
-        assert int(records["Records"][1]["dynamodb"]["SequenceNumber"]) > starting_sequence_number
+        def check_expected_records():
+            records = ddbstreams.get_records(ShardIterator=iterator_id)
```

Comment:
nit: Technically, it could be possible that we're receiving 1 record in the first call to `ddbstreams.get_records(..)`, and then in the next retry iteration again receive 1 record (which would make the assertion in line 805 fail).

I'd suggest that we instantiate a `records = []` list outside the `check_expected_records()` function, and then keep adding records to it - something like:
```
records = []
def check_expected_records():
    records_result = ddbstreams.get_records(ShardIterator=iterator_id)
    records.extend(records_result["Records"])
    ...
```

Reply:
Actually, ignore the comment above - since we're not updating the `ShardIterator` and always passing in the same `iterator_id`, we should be good! 👍